### PR TITLE
Added support for Shopify Collections (both smart and custom).

### PR DESCRIPTION
**Pull request by TonyDeStefano**
Signed-off-by: Tony DeStefano <tdestefano@funko.com>

(Thanks for sending a pull request! Please make sure you click the link above to view the contribution guidelines, then fill out the blanks below.)

What does this implement/fix? Explain your changes.
---------------------------------------------------
…

Does this close any currently open issues?
------------------------------------------
…

Any relevant logs, error output, etc?
-------------------------------------
(If it’s long, please paste to https://ghostbin.com/ and insert the link here.)

Any other comments?
-------------------
…

Where has this been tested?
---------------------------
**Operating System:** …

**Platform:** …

**Version:** …

**Node/PHP/Java Version:** …

### CODE REVIEW

File: src/services/ShopifyService.php [hunk 1]
```diff
@@ -39,40 +78,133 @@ public function getProductsCount($options = array())
...
+        }
+
+        $query = http_build_query($options);
+        $url = $this->getShopifyUrl($endpoint . '?' . $query);
+
+        try {
+            $client = new Client();
+            $response = $client->request('GET', $url);
+
+            if ($response->getStatusCode() !== 200) {
```
Comment (ryanpcmcquen) [inline src/services/ShopifyService.php:186]: @TonyDeStefano what about other 2xx status codes?

File: src/templates/_select.twig [hunk 2]
```diff
@@ -21,114 +22,123 @@
...
+                        {{ option.label }}
+                    </option>
+                {% endif %}
             {% endfor %}
         </select>
     </div>
 
     {% if filter_class is defined and filter_class %}
-        <input style="width: 99.5%;" placeholder="{{ 'Start typing to filter products ...'|t }}" type="text" class="{{ filter_class }}_{{ now }}" value="">
+        <input style="width: 99.5%;" placeholder="{{ 'Start typing to filter ...'|t }}" type="text"
```
Comment (ryanpcmcquen) [inline src/templates/_select.twig]: These need to be `|t('shopify')`.

File: src/templates/_select.twig [hunk 3]
```diff
@@ -21,114 +22,123 @@
...
 
-    attach_events();
+    attach_events()
     document.addEventListener('DOMContentLoaded', () => {
-        attach_events();
-    });
-})();
+      attach_events()
+    })
+  })()
```
Comment (ryanpcmcquen) [inline src/templates/_select.twig]: Can we keep this at 4 spaces with semicolons? I know we use StandardJS normally, but this repo doesn't follow that convention and neither does our Craft code.

File: src/fieldtypes/CollectionFieldType.php [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    }
+
+    /**
+     * @param int $count
+     * @param array $collectionsData
+     * @param string $key
+     * @param array $defaultOptions
+     * @return array
+     */
+    private function _getCollections($count, $collectionsData, $key, $defaultOptions)
```
Comment (ryanpcmcquen) [inline src/fieldtypes/CollectionFieldType.php:102]: Shouldn't these be `public` so the user can make their own field in the twig templates as mentioned in the upstream documentation?

Comment (TonyDeStefano) [reply src/fieldtypes/CollectionFieldType.php:102]: The private functions are only helper functions specific to that class. That was me OOP-ing stuff. The public functions in this class remain as they were.

File: src/templates/_select.twig (hunk 3 above)
Comment (TonyDeStefano) [reply src/templates/_select.twig]: Yeah, I'll do that.

File: src/templates/_select.twig (hunk 2 above)
Comment (TonyDeStefano) [reply src/templates/_select.twig]: Weird, I feel like I just copy-pasted.

File: src/services/ShopifyService.php (hunk 1 above)
Comment (TonyDeStefano) [reply src/services/ShopifyService.php:186]: I only moved that code. I didn't change it at all.

File: src/templates/_select.twig (hunk 2 above)
Comment (ryanpcmcquen) [reply src/templates/_select.twig]: It may be wrong in my branch.

File: src/services/ShopifyService.php (hunk 1 above)
Comment (ryanpcmcquen) [reply src/services/ShopifyService.php:186]: 😢

File: src/fieldtypes/CollectionFieldType.php (hunk 4 above)
Comment (ryanpcmcquen) [reply src/fieldtypes/CollectionFieldType.php:102]: Ah, I see. I don't know how to read code.

File: src/services/ShopifyService.php (hunk 1 above)
Comment (ryanpcmcquen) [reply src/services/ShopifyService.php:186]: Maybe we should switch it to `isOk` or whatever this lib has?

**Review by ryanpcmcquen — APPROVED** [review]
Nice work!

**Pull request merged**